src/sort/sortingScript.sml

```diff
@@ -1445,6 +1445,24 @@ Proof
   \\ PROVE_TAC[]
 QED
 
+Theorem LIST_REL_PERM:
```

Comment:
- Depends on generated name `M`
- seems to be expressed to suit `ho_match_mp_tac`

Suggest:
```
Theorem LIST_REL_PERM:
  !l1 l2 l3. LIST_REL P l1 l2 /\ PERM l2 l3 ==>
             ?l4. PERM l1 l4 /\ LIST_REL P l4 l3
Proof
  Induct_on ‘LIST_REL’
  \\ rw[PERM_CONS_EQ_APPEND, PULL_EXISTS]
  \\ ntac 2 (irule_at Any listTheory.LIST_REL_APPEND_suff)
  \\ simp[]
  \\ first_x_assum $ drule_then strip_assume_tac
  \\ gs[LIST_REL_SPLIT2, SF SFY_ss]
QED
```

Reply:
Very nice! Done.